Modules/Common/AptosLikeMainModule.php

```diff
@@ -0,0 +1,297 @@
+
+/*  This module process the APT transfers in Aptos Blockchain. */
+
+abstract class AptosLikeMainModule extends CoreModule
+{
+    use AptosTraits;
+
+    public ?BlockHashFormat $block_hash_format = BlockHashFormat::HexWith0x;
+    public ?AddressFormat $address_format = AddressFormat::HexWith0x;
+    public ?TransactionHashFormat $transaction_hash_format = TransactionHashFormat::HexWith0x;
+    public ?TransactionRenderModel $transaction_render_model = TransactionRenderModel::Even;
+    public ?CurrencyFormat $currency_format = CurrencyFormat::Static;
+    public ?CurrencyType $currency_type = CurrencyType::FT;
+    public ?FeeRenderModel $fee_render_model = FeeRenderModel::ExtraF;
+    public ?array $special_addresses = ['the-void', 'the-contract'];
+    public ?PrivacyModel $privacy_model = PrivacyModel::Transparent;
+    public ?bool $ignore_sum_of_all_effects = false;
+
+    public ?array $events_table_fields = ['block', 'transaction', 'sort_key', 'time', 'address', 'effect', 'failed', 'extra'];
+    public ?array $events_table_nullable_fields = ['extra'];
+
+    public ?ExtraDataModel $extra_data_model = ExtraDataModel::Type;
+    public ?array $extra_data_details = [AptosSpecialTransactions::ValidatorReward->value => 'Validator reward',];
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = false;
+    public ?bool $allow_empty_return_events = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = false;
+
+    final public function pre_initialize()
+    {
+        $this->version = 1;
+    }
+
+    final public function post_post_initialize()
+    {
+
+    }
+
+    final public function pre_process_block($block_id)
+    {
+        $block = requester_single($this->select_node(), endpoint: "v1/blocks/by_height/{$block_id}?with_transactions=true", timeout: $this->timeout);
+        $this->block_time = date('Y-m-d H:i:s', (int) ((int) $block['block_timestamp'] / 1000000));
+
+        $events = [];
+        $sort_key = 0;
+        foreach ($block['transactions'] as $trx)
+        {
+            $failed = false;
+            if ($trx['vm_status'] !== 'Executed successfully')
+            {
+                $failed = true;
+            }
+
+            switch ($trx['type'])
+            {
+                case 'genesis_transaction':
+                    // Process initial APT balances in changes at genesis block.
+                    foreach ($trx['changes'] as $change)
+                    {
+                        $changed_resource = $change['data']['type'] ?? null;
+                        if ($changed_resource !== '0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>')
+                        {
+                            continue;
+                        }
+
+                        $address = $change['address'];
+                        $balance = $change['data']['data']['coin']['value'] ?? '0';
+                        if ($balance === '0')
```

Comment:
Generally, we track zero transfers too.

Reply:
In this case `genesis_transactions` processed. `if ($balance === '0')` for checking resources with only created resources after genesis without any APT in this resources.